### DISCUSSION TITLE
Testing install of zed

### DISCUSSION THREAD

**Discussion by lundman** [openzfsonwindows/openzfs#418] [Development]
https://github.com/openzfsonwindows/openzfs/releases/download/zfswin-2.2.6rc10/OpenZFSOnWindows-debug-.99-3126-g97bdecb6f.exe

For those that can, testing out `zed` running as a service. 

Hopefully, the driver should install as usual, then it installs `zed`. You can verify that the install went ok;

```
sc query OpenZFS_zed
```

Which should list the service, and hopefully be in `RUNNING` state.

It should write debug log to `C:/Program Files/OpenZFS on Windows/zed.txt` - but that is just for now.

You are supposed to be able to open up `Event Viewer`, under `Event Viewer (local) / Windows Logs / Application` is where `OpenZFS_zed` events should show, if there are any.

`zed` uses the `zedlets` (rewritten in powershell for Windows) under `C:/Program Files/OpenZFS on Windows/zed.d/` - currently only really `all-debug.ps1` and `system_boot.ps1` are tested. 

If you have (at least one) pool imported, you should have a `C:/Windows/System32/drivers/zpool.cache` so if you reboot, `system_boot.ps1` runs, and if `zpool.cache` is present, it runs `zpool import` on it, and in theory, your pool(s) are imported again.

Please let me know if you come across any quirkiness.

**Comment by guenther-alka** [comment]
I updated rc10 to 3126 (with installed Aomei)
This ended with a BSOD (kernel security check)
Zed is running after reboot

memory.dmp if you need
https://drive.google.com/file/d/1-txh3V_-1Qadt9-rzKv49gL83p-TBoB3/view?usp=sharing

I then deinstalled Aomei as I suppose this was the problem and run the installer again.
Installer complains about not be able to replace zed.exe. I ignored and installation went through, 
reboot manually (ok), zed running

```
PS C:\Users\me> get-process -name zed

Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName
-------  ------    -----      -----     ------     --  -- -----------
    132       8     3252       6632       0,06   2952   0 zed
```

**Reply by lundman** [reply]
Oh hmm right, so installing should probably look to stop `zed` as well.  Otherwise it went ok? Pool got imported?

**Comment by guenther-alka** [comment]
My testpool (on an USB stick) is not auto imported after reboot. 
When I import manually i see the C:/Windows/System32/drivers/zpool.cache

**Reply by lundman** [reply]
And you just reboot yes? If you export, it will be removed from zpool.cache.

**Comment by guenther-alka** [comment]
I just rebooted with zpool.cache in ./drivers

**Reply by lundman** [reply]
Anything in Event Viewer? Or indeed C:/Program Files/OpenZFS on Windows/zed.txt

**Comment by guenther-alka** [comment]
I found two application events
```
2024-11-27 09:17:17 [LOG_NOTICE] zpool.cache found. Importing ZFS pools
2024-11-27 09:17:23 [ZFS Pools Imported.] 
```
but pool not imported
C:/Program Files/OpenZFS on Windows/zed.txt not created.

If it is important
I have Avast Antivir installed

**Reply by lundman** [reply]
Well, sounds like `zed` installed ok, and is running, and is even .. you know.. pretending to help.  

`https://github.com/openzfsonwindows/openzfs/blob/zed/cmd/zed/os/windows/zed.d/system_boot.ps1`

all it is supposed to do next is run

` & "$env:ZPOOL" import -c "$zpoolCachePath" -a`

But no idea what went wrong there, not like it has output

**Comment by guenther-alka** [comment]
ok, my fault
I had a testpool on .vhdx that is automounted but after zed so zed cannot mount

I recreated a testpool on usb and automount works now

**Reply by lundman** [reply]
Ah heh yeah, it doesn't know to mount the vhd - although, you could add that line to `system_boot.ps1` if you want to

**Comment by guenther-alka** [comment]
Automount, (autoimport) and autounlock is already done when my web-gui starts manually or as a planned task

**Comment by guenther-alka** [comment]
Update from rc10 with zed installed to rc11 gives an error "could not overwrite zed.exe".
I ignored and update went through with old zed.exe.
